notebooks/15_flightplanning_tuto.ipynb

```diff
@@ -0,0 +1,1018 @@
+   "metadata": {},
+   "source": [
+    "In the following, we will be using LazyAstar (A*) which is a solver coming from the path planning community."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Solver instantiation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As explained earlier, we use the Boundary Extension Encoding state features `bee2_features` so that IW can dynamically increase state variable intervals. In other domains, other state features might be more suitable."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "solver = Astar(heuristic=lambda d, s: d.heuristic(s))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Training solver on domain"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "FlightPlanningDomain.solve_with(solver, domain_factory)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Rolling out a solution\n",
+    "\n",
+    "**Disclaimer:** This roll out can be a bit painful to look on local Jupyter sessions. Indeed, IW creates copies of the environment at each step which makes pop up then close a new OpenGL window each time."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We have to slightly modify the roll out function as observations for the new domain are now wrapped in a `GymDomainProxyState` to make them serializable. So to get access to the underlying numpy array, we need to look for `observation._state`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pause_between_steps = None\n",
+    "max_steps = 100\n",
+    "\n",
+    "solver.reset()\n",
+    "observation = domain.reset()\n",
+    "\n",
+    "print(\"Starting planning\")\n",
+    "\n",
+    "# Initialize image\n",
+    "figure = domain.render(observation)\n",
+    "display(figure)\n",
+    "\n",
+    "# loop until max_steps or goal is reached\n",
+    "for i_step in range(1, max_steps + 1):\n",
+    "    if pause_between_steps is not None:\n",
+    "        sleep(pause_between_steps)\n",
+    "\n",
+    "    # choose action according to solver\n",
+    "    action = solver.sample_action(observation)\n",
+    "    # get corresponding action\n",
+    "    outcome = domain.step(action)\n",
+    "    observation = outcome.observation\n",
+    "\n",
+    "    # update image\n",
+    "    figure = domain.render(observation)\n",
+    "    clear_output(wait=True)\n",
+    "    display(figure)\n",
+    "\n",
+    "    # final state reached?\n",
+    "    if domain.is_terminal(observation):\n",
+    "        break\n",
+    "\n",
+    "# goal reached?\n",
+    "is_goal_reached = domain.is_goal(observation)\n",
```

Comment:
rendering in my jupyter session just appeared at the end of the rollout (which is fine but maybe unexpected)

Reply:
fixed